src/runner/Run_semgrep.ml

```diff
@@ -568,6 +568,34 @@ let xtarget_of_file (config : Runner_config.t) (xlang : Xlang.t)
     lazy_ast_and_errors;
   }
 
+(* [HACK "targets hash consing"]
+ * Since most rules do not define any 'includes'/'excludes', the 'rule_nums'
```

Comment:
the include/exclude are actually handled now in semgrep-core ... so we definitely don't need anymore this complex Input_to_core.atd data structure

Reply:
See the call to Filter_target.filter_paths in Run_semgrep.ml

Reply:
It was just an experiment but since it's just one function and easy to remove without breaking anything, I thought it could be useful to merge if that may help users with large monorepos. But assuming that our per-rule include/exclude is equivalent to that of pysemgrep, then we could just send Semgrep-core the list of targets per language, so we would just have:

```ocaml
type target <ocaml attr="deriving show"> = {
  path: string (* source file *);
  (* Must be a valid "language" as defined in Xlang.mli.
     examples: "ocaml", "python", but also "spacegrep" or "regexp".
     less: would be great to use Lang.t here, but atd does not
     allow (by design) to use types defined in other files. *)
  language: xlang;
}
```

And semgrep-core will decide what rules run on each target. Is that what you are thinking of as the right thing?

Reply:
Ideally yes; it's worth trying ... but maybe it's more complicated than I think to remove this rul_num field
and we get some regressions.

Reply:
@IagoAbal if you want to help a little bit in our effort to get file targeting fully ocaml-side by Q3, this is a good way to help :)

Reply:
Wait it's handled now regardless of whether you call it with `osemgrep` or not?

Reply:
correct.
there's even a comment about it.

Reply:
Let's get rid of Input_to_core!!!!!!!!!!

Reply:
(How much do we trust it? Do we need `--legacy` to still use Input_to_core?)